src/middlewared/middlewared/api/v25_10_0/pool.py

```diff
@@ -21,6 +23,20 @@
 ]
 
 
+# Incus cannot consume a pool which has whitespaces in its name.
+# FIXME: Once this is fixed on incus side, we can remove this and keep on relying libzfs to do the validation only
+POOL_NAME: TypeAlias = Annotated[
+    NonEmptyString,
+    AfterValidator(
+        match_validator(
+            re.compile(r"^\S+$"),
```

Comment:
Should we just whitelist allowed characters? `[a-z][a-z0-9_.\-]*`

Reply:
That validation lives in libzfs atm and it is being consumed from there, hence i did not want to specify that in the regex itself